ethcore/src/miner/miner.rs

```diff
@@ -117,267 +110,185 @@ pub struct MinerOptions {
-		if Instant::now() >= self.next_calibration {
-			let usd_per_tx = self.options.usd_per_tx;
-			trace!(target: "miner", "Getting price info");
-
-			self.price_info.get(move |price: PriceInfo| {
-				trace!(target: "miner", "Price info arrived: {:?}", price);
-				let usd_per_eth = price.ethusd;
-				let wei_per_usd: f32 = 1.0e18 / usd_per_eth;
-				let gas_per_tx: f32 = 21000.0;
-				let wei_per_gas: f32 = wei_per_usd * usd_per_tx / gas_per_tx;
-				info!(target: "miner", "Updated conversion rate to Ξ1 = {} ({} wei/gas)", Colour::White.bold().paint(format!("US${:.2}", usd_per_eth)), Colour::Yellow.bold().paint(format!("{}", wei_per_gas)));
-				set_price(U256::from(wei_per_gas as u64));
-			});
-
-			self.next_calibration = Instant::now() + self.options.recalibration_period;
-		}
-	}
-}
-
-/// Struct to look after updating the acceptable gas price of a miner.
-#[derive(Debug, PartialEq)]
-pub enum GasPricer {
-	/// A fixed gas price in terms of Wei - always the argument given.
-	Fixed(U256),
-	/// Gas price is calibrated according to a fixed amount of USD.
-	Calibrated(GasPriceCalibrator),
+struct SealingWork {
+	queue: UsingQueue<ClosedBlock>,
+	enabled: bool,
+	next_allowed_reseal: Instant,
+	next_mandatory_reseal: Instant,
+	// block number when sealing work was last requested
+	last_request: u64,
 }
 
-impl GasPricer {
-	/// Create a new Calibrated `GasPricer`.
-	pub fn new_calibrated(options: GasPriceCalibratorOptions, fetch: FetchClient) -> GasPricer {
-		GasPricer::Calibrated(GasPriceCalibrator {
-			options: options,
-			next_calibration: Instant::now(),
-			price_info: PriceInfoClient::new(fetch),
-		})
-	}
-
-	/// Create a new Fixed `GasPricer`.
-	pub fn new_fixed(gas_price: U256) -> GasPricer {
-		GasPricer::Fixed(gas_price)
-	}
-
-	fn recalibrate<F: Fn(U256) + Sync + Send + 'static>(&mut self, set_price: F) {
-		match *self {
-			GasPricer::Fixed(ref max) => set_price(max.clone()),
-			GasPricer::Calibrated(ref mut cal) => cal.recalibrate(set_price),
-		}
+impl SealingWork {
+	/// Are we allowed to do a non-mandatory reseal?
+	fn reseal_allowed(&self) -> bool {
+		Instant::now() > self.next_allowed_reseal
 	}
-}
 
-struct SealingWork {
-	queue: UsingQueue<ClosedBlock>,
-	enabled: bool,
 }
 
 /// Keeps track of transactions using priority queue and holds currently mined block.
 /// Handles preparing work for "work sealing" or seals "internally" if Engine does not require work.
 pub struct Miner {
 	// NOTE [ToDr]  When locking always lock in this order!
-	transaction_queue: Arc<RwLock<BanningTransactionQueue>>,
-	transaction_listener: RwLock<Vec<Box<Fn(&[H256]) + Send + Sync>>>,
-	sealing_work: Mutex<SealingWork>,
-	next_allowed_reseal: Mutex<Instant>,
-	next_mandatory_reseal: RwLock<Instant>,
-	sealing_block_last_request: Mutex<u64>,
-	// for sealing...
+	sealing: Mutex<SealingWork>,
+	params: RwLock<AuthoringParams>,
+	listeners: RwLock<Vec<Box<NotifyWork>>>,
+	gas_pricer: Mutex<GasPricer>,
```

Comment:
Looks like this part may subject to refactoring using the idea mentioned in https://github.com/paritytech/parity/issues/7079